### DISCUSSION TITLE
WebApp installation on DIRAC 8.0.0a26

### DISCUSSION THREAD

**Discussion by IgorPelevanyuk** [DIRACGrid/DIRAC#6276] [Q&A] [answered]
Hi, 
I have issue with WebApp installation during DIRAC 8.0.0a26 installation. 
The error is the following:
`ERROR: Cannot install dirac[server]==8.0.0a26, webappdirac[server]==4.3.1, webappdirac[server]==4.3.10, webappdirac[server]==4.3.11, webappdirac[server]==4.3.2, webappdirac[server]==4.3.3, webappdirac[server]==4.3.4, webappdirac[server]==4.3.5, webappdirac[server]==4.3.6, webappdirac[server]==4.3.7, webappdirac[server]==4.3.8 and webappdirac[server]==4.3.9 because these package versions have conflicting dependencies.`

Configuration related to WebApp installation:
```
Extensions = WebApp
WebApp = yes
WebPortal = yes
```
Can it be that for DIRAC 8 installation of WebApp is somehow different?

Kind regards,
Igor Pelevanyuk

**Comment by fstagni** [comment, marked as answer]
Looks like it is trying to install every webApp version but the correct one.
From which version are you installing, and how?
I guess you already migrated to python3 ...

**Reply by IgorPelevanyuk** [reply]
I am planning to install fresh DIRAC 8 installation parallely with currently running in production DIRAC 7. When DIRAC 8 works propperly, just migrate AccountingDB and FileCatalogDB.
I was using documentation on DIRAC installation from here https://dirac.readthedocs.io/en/latest/AdministratorGuide/ServerInstallations/InstallingDiracServer.html
To install dirac I execute command: 
`./install_site.sh -i /opt/dirac -v 8.0.0a26 /home/dirac/DIRAC/install.cfg`
install.cfg:
```
LocalInstallation
{
  VirtualOrganization = JINR
  SiteName = JINR-LIT
  Setup = JINR-Production
  InstanceName = Production
  SkipCADownload = yes
  UseServerCertificate = yes
  ConfigurationName = Dirac-Prod
  AdminUserName = ipelevanyuk
  AdminUserDN = /C=RU/O=RDIG/OU=users/OU=jinr.ru/CN=my cert
  AdminUserEmail = email@gmail.com
  AdminGroupName = dirac_admin
  HostDN = /C=RU/O=RDIG/OU=hosts/OU=jinr.ru/CN=hostname.jinr.ru
  ConfigurationMaster = yes
  Extensions = WebApp
  Extensions += VMDIRAC
  Extensions += COMDIRAC

  Host = hostname.jinr.ru
  Databases = InstalledComponentsDB
  Databases += ResourceStatusDB
  Databases += SandboxMetadataDB
  Databases += JobDB
  Databases += FileCatalogDB
  Databases += JobLoggingDB
  Databases += UserProfileDB
  Databases += TaskQueueDB
  Databases += NotificationDB
  Databases += ReqDB
  #Databases += FTSDB
  Databases += ProxyDB
  Databases += PilotAgentsDB
  Databases += AccountingDB
  Databases += TransformationDB

  Services  = Configuration/Server
  Services += Framework/ComponentMonitoring
  Services += Framework/SystemAdministrator
  Services += DataManagement/StorageElement
  Services += DataManagement/FileCatalog
  Services += DataManagement/StorageElementProxy
  Services += DataManagement/FTSManager
  Services += Framework/Monitoring
  Services += Framework/Notification
  Services += Framework/SecurityLogging
  Services += Framework/UserProfileManager
  Services += Framework/ProxyManager
  Services += Framework/Plotting
  Services += Framework/BundleDelivery
  Services += Monitoring/Monitoring
  Services += WorkloadManagement/SandboxStore
  Services += WorkloadManagement/Matcher
  Services += WorkloadManagement/JobMonitoring
  Services += WorkloadManagement/JobManager
  Services += WorkloadManagement/JobStateUpdate
  Services += WorkloadManagement/WMSAdministrator
  Services += WorkloadManagement/OptimizationMind
  Services += RequestManagement/ReqManager
  Services += Accounting/DataStore
  Services += Accounting/ReportGenerator
  Services += ResourceStatus/ResourceManagement
  Services += ResourceStatus/ResourceStatus
  Services += Transformation/TransformationManager

  Agents  = Framework/CAUpdateAgent
  Agents += DataManagement/FTS3Agent
  Agents += WorkloadManagement/PilotStatusAgent
  Agents += WorkloadManagement/SiteDirector
  Agents += WorkloadManagement/JobCleaningAgent
  Agents += WorkloadManagement/StalledJobAgent
  Agents += WorkloadManagement/StatesAccountingAgent
  Agents += Configuration/Bdii2CSAgent
  Agents += Configuration/VOMS2CSAgent
  Agents += RequestManagement/RequestExecutingAgent
  Agents += RequestManagement/CleanReqDBAgent

  Executors = WorkloadManagement/Optimizers

  WebApp = yes
  WebPortal = yes
  # Database configuration MariaDB + Elasticsearch
}
```

**Reply by IgorPelevanyuk** [reply]
I also tried to use 7.3.20. With that version WebApp installs successfully. 
Maybe that is because in releases.cfg version 8.0.26a is absent?
Is that even possible/practical to install DIRAC 8 right now?

**Reply by marianne013** [reply]
v8 is not certified yet - though I am sure Federico will welcome more testers :-)

**Reply by IgorPelevanyuk** [reply]
Thanks Daniela for the information! Looks like right now it is better to install latest DIRAC v7r3p26 and upgrade later.
